### ISSUE TITLE
safeHTML marked HTML data elements (.Site.Data, JSON etc.) are escaped

### ISSUE THREAD

**Issue by Brixy** [gohugoio/hugo#3876]
Thanks for v0.27. Related content is great and fast.

Just want to mention that (some?) shortcodes suddenly escape html. The instagram expample from the website results in:

```
&lt;blockquote class=&#34;instagram-media&#34; data-instgrm-captioned data-instgrm-version=&#34;7&#34; style=&#34; background:#FFF; border:0; border-radius:3px; box-shadow:0 0 1px 0 rgba(0,0,0,0.5),0 1px 10px 0 rgba(0,0,0,0.15); margin: 1px; max-width:658px; padding:0; width:99.375%; width:-webkit-calc(100% - 2px); width:calc(100% - 2px);&#34;&gt;&lt;div style=&#34;padding:8px;&#34;&gt; &lt;div style=&#34; background:#F8F8F8; line-height:0; margin-top:40px; padding:33.251231527093594% 0; text-align:center; width:100%;&#34;&gt; &lt;div style=&#34;

…
```

Also my own shortcodes are escaped.

Thank you.

**Comment by bep** [comment]
Hmm... Probably a Go 1.9 thing. But gohugo.io is built with Hugo 0.27 and I don't see any issue there:

https://gohugo.io/content-management/shortcodes/#instagram

**Comment by bep** [comment]
What OS are you running?

@digitalcraftsman can you reproduce this?

**Comment by Brixy** [comment]
macOS 10.12.6

Thanks for your help, @bep. I guess, your middle name is “fast”, too!

**Comment by bep** [comment]
>Thanks for your help, @bep. I guess, your middle name is “fast”, too!

No, that is my last name. My middle name is Erik :-)

**Comment by digitalcraftsman** [comment]
I compiled the latest version and checked the docs locally. But I wasn't able to reproduce this issue either.

**Comment by Brixy** [comment]
Thank you, guys.

Now, I know what _causes_ the problem, but I do not know why.

My “config.toml” says:

```
[outputs]
page = ["HTML", "Calendar"]
```

It is the `Calendar` entry that causes the error. Without it, everything works fine—but, of course, the calendar function is necessary for the project. With that said, I just love Hugo for its custom output formats!

Error occurs with “single.ics” or with “single.calendar.ics”

**Comment by bep** [comment]
@Brixy OK, then I think I know enough to try to reproduce it. I suspect it is a Go 1.9 issue, which is the only related change from Hugo 0.26.

**Comment by pamubay** [comment]
I also face this issue. 

it happened when running `hugo server`

in `0.26` if we change the shortcode (like Instagram) Hugo rebuild always generate successfully without escaping the html.

in `0.27` when `hugo server` running then we change the shortcode, the rebuild will produce escaped HTML code.

**Comment by bep** [comment]
See https://github.com/golang/go/issues/21844

**Comment by Brixy** [comment]
Thank you very much for all your efforts in general and for bringing out v0.27.1 so quickly, @bep.

(Apologies if answering to a closed thread is unprofessional[?].)

**Comment by bep** [comment]
@Brixy the reason I moved so fast on this was that I suspected that it had to be fixed in Go, and I knew that 1.9.1 is just around the corner -- so to get a fix in there would require some really fast action. Luckily there was a workaround on the Hugo side of the fence.
